platform/qt/src/settings.cpp

```diff
@@ -0,0 +1,411 @@
+
+    By default, it is set to QMapLibreGL::Settings::ConstrainHeightOnly.
+*/
+Settings::ConstrainMode Settings::constrainMode() const
+{
+    return m_constrainMode;
+}
+
+/*!
+    Sets the map constrain \a mode.
+*/
+void Settings::setConstrainMode(ConstrainMode mode)
+{
+    m_constrainMode = mode;
+}
+
+/*!
+    Returns the viewport mode. This is used to flip the vertical
+    orientation of the map as some devices may use inverted orientation.
+
+    By default, it is set to QMapLibreGL::Settings::DefaultViewport.
+*/
+Settings::ViewportMode Settings::viewportMode() const
+{
+    return m_viewportMode;
+}
+
+/*!
+    Sets the viewport \a mode.
+*/
+void Settings::setViewportMode(ViewportMode mode)
+{
+    m_viewportMode = mode;
+}
+
+/*!
+    Returns the cache database maximum hard size in bytes. The database
+    will grow until the limit is reached. Setting a maximum size smaller
+    than the current size of an existing database results in undefined
+    behavior
+
+    By default, it is set to 50 MB.
+*/
+unsigned Settings::cacheDatabaseMaximumSize() const
+{
+    return m_cacheMaximumSize;
+}
+
+/*!
+    Returns the maximum allowed cache database \a size in bytes.
+*/
+void Settings::setCacheDatabaseMaximumSize(unsigned size)
+{
+    m_cacheMaximumSize = size;
+}
+
+/*!
+    Returns the cache database path. The cache is used for storing
+    recently used resources like tiles and also an offline tile database
+    pre-populated by the
+    \l {https://github.com/mapbox/mapbox-gl-native/blob/master/bin/offline.sh}
+    {Offline Tool}.
+
+    By default, it is set to \c :memory: meaning it will create an in-memory
+    cache instead of a file on disk.
+*/
+QString Settings::cacheDatabasePath() const
+{
+    return m_cacheDatabasePath;
+}
+
+/*!
+    Sets the cache database \a path.
+
+    Setting the \a path to \c :memory: will create an in-memory cache.
+*/
+void Settings::setCacheDatabasePath(const QString &path)
+{
+    m_cacheDatabasePath = path;
+}
+
+/*!
+    Returns the asset path, which is the root directory from where
+    the \c asset:// scheme gets resolved in a style. \c asset:// can be used
+    for loading a resource from the disk in a style rather than fetching
+    it from the network.
+
+    By default, it is set to the value returned by QCoreApplication::applicationDirPath().
+*/
+QString Settings::assetPath() const
+{
+    return m_assetPath;
+}
+
+/*!
+    Sets the asset \a path.
+*/
+void Settings::setAssetPath(const QString &path)
+{
+    m_assetPath = path;
+}
+
+/*!
+    Returns the API key.
+
+    By default, it is taken from the environment variable \c MGL_API_KEY
+    or empty if the variable is not set.
+*/
+QString Settings::apiKey() const {
+    return m_apiKey;
+}
+
+/*!
+    Sets the API key.
+
+    MapTiler-hosted and Mapbox-hosted vector tiles and styles require an API
+    key or access token.
+*/
+void Settings::setApiKey(const QString &key)
+{
+    m_apiKey = key;
+}
```

Comment:
General question: Is native still working with mapbox api keys? We removed them in GL JS in the TypeScript migration.

Reply:
No idea actually. Would it break because of MapBox API changes or just because we do not want to support it?

Reply:
I think MapLibre should not have to support Mapbox API keys. If someone uses their API, they also should use their library. Mapbox are doing a great job and deserve their customers...

Reply:
OK, let me clean this up in a follow-up MR, maybe even fully for the whole core repository. Will open a ticket for it.

BTW what about MapTiler? I'm not in the loop how they handle API keys.

Reply:
Let's continue the discussion on #349 (and probably the steering meeting).

Reply:
Yes, it does work with Mapbox keys. And I hope it will stay that way, as some users prefer Mapbox for maps. 

This comment is based on the state of the repository from few months ago. I hope that hasn't been changed

Reply:
PS: Actually, from testing just now with Mapbox keys - all looks to be fine.